js/data/buffer/collision_box_vertex_buffer.js

```diff
@@ -1,37 +1,34 @@
 'use strict';
 
 var util = require('../../util/util');
-var Buffer = require('./buffer');
-
-module.exports = CollisionBoxVertexBuffer;
-
-function CollisionBoxVertexBuffer(buffer) {
-    Buffer.call(this, buffer);
+var Buffer2 = require('../buffer2');
+
+function CollisionBoxVertexBuffer(options) {
+    Buffer2.call(this, options || {
+        type: Buffer2.BufferType.VERTEX,
+        attributes: {
```

Comment:
The `Buffer2` class allows us to express the layout in a
more straightforward and self-documenting way. I'd write this class' attributes as

```js
attributes: {
  coordinates: {
    type: Buffer2.AttributeType.SIGNED_SHORT,
    components: 2
  },
  extrusions: {
    type: Buffer2.AttributeType.SIGNED_SHORT,
    components: 2
  },
  maxZoom: {
    type: Buffer2.AttributeType.UNSIGNED_BYTE,
  }
  placementZoom: {
    type: Buffer2.AttributeType.UNSIGNED_BYTE,
  }
}
```